.changeset/wild-geckos-draw.md

```diff
@@ -0,0 +1,18 @@
+---
+'astro': patch
+---
+
+Keep clientAddress on cloned requests
+
+User observed that calling actions resulted in an error about not having
+clientRequest available.
+
+This is because the user had a middleware that cloned the request, which
+loses all of the symbols.
+
+The fix is to pass the clientAddress directly into the RenderContext.
+This deprecates the `clientAddressSymbol`, but we need to keep it for
+now because some adapters set the clientAddress that way.
+
+Note that similar fixes should be done for other symbol usage on the
+Request object (locals is one).
```

Comment:
Can we reword the changeset for the users? This one contains a lot of information that users might not understand or don't need. 

We might want to highlight the actual deprecation though, and maybe **log a warning** if the user attempts to use the old value

Reply:
Will reword the changeset. I don't think we should start spitting out warnings this early in a major though. We definitely at least want to update all of the adapters first to not set this symbol.